### PR TITLE
Add feature to import keepass binary attachments

### PR DESCRIPTION
**Pull request by zioalex**
Using the pykeepass library I get the attachment list.

### CODE REVIEW

**Comment by roddhjav** [comment]
Hi, Thank for this PR. I will certainly add this feature to pass-import. However, there are a few issues with your proposal.

- A lot of your changes have nothing to do with your PR. They are linter changes that could break the existing linter style in the project. See `make lint` and `.prospector.yml`.
- It is not secure to read a password from an argument. You should do it directly in `Keepass.parse()` with [getpass](https://docs.python.org/3/library/getpass.html)
- The `-b` option is not required.
- Now https://github.com/pschmitt/pykeepass/pull/126 has been merged, you should maybe use it in `pass-import`.
- As of today, binary attachments is only supported for Keepass, therefore your changes should mostly be in the Keepass class. Therefore the code in PasswordStore.insert_binary should probably be moved in Keepass.

This is only a first fast look on you PR. I will have a deeper look later.

**Comment by zioalex** [comment]
If you think is worthy to have it merged I can work to improve it .

**Comment by roddhjav** [comment]
Yes, it is worthy to merge it.

File: pass_import.py [hunk 1]
```diff
@@ -32,6 +32,8 @@
 
 __version__ = '2.4'
 
+TMP_ATTACHMENT_DIR = "tmp_attachment_files/"
```
Comment (roddhjav) [inline pass_import.py]: This should not be static. You should use a secure temporary directory of the user with https://docs.python.org/3/library/tempfile.html. And therefore, it does not have to be a global variable.

File: pass_import.py [hunk 2]
```diff
@@ -184,11 +186,32 @@ def insert(self, path, data, force=False):
         """Multiline insertion into the password store."""
         if not force:
             if os.path.isfile(os.path.join(self.prefix, path + '.gpg')):
-                raise PasswordStoreError("An entry already exists for %s." % path)
+                raise PasswordStoreError(
```
Comment (roddhjav) [inline pass_import.py]: Not in the scope of this PR. (Same for all similar changes)

BTW, the max-line-length in this project is 99 anyway. See `.prospector.yml`

File: pass_import.py [hunk 3]
```diff
@@ -255,19 +280,21 @@ def _clean_protocol(self, string):
...
         """Convert invalid caracters by the separator in a string."""
-        caracters = dict(zip(self.invalids, [self.separator]*len(self.invalids)))
+        caracters = dict(
+            zip(self.invalids, [self.separator]*len(self.invalids)))
         return self._replaces(caracters, string)
 
     def _clean_cmdline(self, string):
         """Make the string more command line friendly."""
-        return self._replaces(self.cleans, string)
+        return self._replaces(self.cleans, string.strip())
```
Comment (roddhjav) [inline pass_import.py]: Not in the scope. Furthermore, the purpose of this function, when called is to replace spaces, not to remove them.

File: pass_import.py [hunk 4]
```diff
@@ -735,9 +863,9 @@ def argumentsparse(argv):
   Import data from most of the password manager. Passwords
   are imported in the existing default password store, therefore
   the password store must have been initialised before with 'pass init'""",
-    usage="%(prog)s [-h] [-V] [[-p PATH] [-c] [-C] [-s] [-e] [-f] | -l] [manager] [file]",
-    formatter_class=argparse.RawDescriptionHelpFormatter,
-    epilog="More information may be found in the pass-import(1) man page.")
+                                     usage="%(prog)s [-h] [-V] [[-p PATH] [-c] [-C] [-s] [-e] [-b              KeePassFile password][-f] | -l] [manager] [file]",
```
Comment (roddhjav) [inline pass_import.py]: The `-b` option is not needed. If there are binary files, it is needed to import them.

File: pass_import.py [hunk 5]
```diff
@@ -833,6 +965,16 @@ def report(arg, msg, paths):
         paths.sort()
         for path in paths:
             msg.echo(os.path.join(arg.root, path))
+    if binaries:
```
Comment (roddhjav) [inline pass_import.py]: A binary import is an import like any other, so you don't have to report it differently.

File: pass_import.py [hunk 6]
```diff
@@ -613,6 +658,86 @@ def _getpath(cls, element, path=''):
...
+                bin_elem = keepassobj._xpath(
+                    '/KeePassFile/Meta/Binaries/Binary[@ID=' + binary_ref + ']')[0]
+
+                path = TMP_ATTACHMENT_DIR + self._replaces(self.cleans, entry.title)
+                if not os.path.exists(path):
+                    os.makedirs(path)
+                self.write_attach(bin_elem, attachment_name, path)
+
+
+    def parse(self, file, password=None):
```
Comment (roddhjav) [inline pass_import.py]: `password` should be read from user directly in this parse method.

**Comment by roddhjav** [comment]
Closed due to PR issue and lack of feedback. Feature added in #84.

**Pull request closed without merging**